src/app/pref/preferences.cpp

```diff
@@ -139,7 +139,20 @@ bool Preferences::isSet(OptionBase& opt) const
 
 ToolPreferences& Preferences::tool(tools::Tool* tool)
 {
-  ASSERT(tool != NULL);
+  // If tool == nullptr it means that the tool's options are shared with all tools.
+  if (!tool) {
+    auto it = m_tools.find("shared");
+    if (it != m_tools.end()) {
+      return *it->second;
+    }
+    else {
+      std::string section = std::string("tool.shared");
+      ToolPreferences* toolPref = new ToolPreferences(section);
+      m_tools["shared"] = toolPref;
+      toolPref->load();
+      return *toolPref;
+    }
+  }
```

Comment:
As this code is similar to the one that follows, probably we can create an `id` local variable and do something like this:

```suggestion
 std::string id;
 
  // If tool == nullptr it means that the tool's options are shared with all tools.
  if (tool)
    id = tool->getId();
  else
    id = "shared";
```

and use `id` instead of `tool->getId()` in the following code, and check for `tool` before using `tool->`.